lib/fake-xhr/index.test.js

```diff
@@ -1413,6 +1428,112 @@ describe("FakeXMLHttpRequest", function () {
         });
     });
 
+    describe(".triggerTimeout", function () {
+        beforeEach(function () {
+            this.sandbox = sinonSandbox.create();
+            this.xhr = new FakeXMLHttpRequest();
+        });
+
+        afterEach(function () {
+            this.sandbox.restore();
+        });
+
+        it("sets timedOut to true", function () {
+            this.xhr.triggerTimeout();
+
+            assert.isTrue(this.xhr.timedOut);
+        });
+
+        it("does not get called without a fake timer", function (done) {
+            var xhr = this.xhr;
+
+            xhr.open("GET", "/");
+            xhr.timeout = 1;
+            xhr.triggerTimeout = sinonSpy();
+            xhr.send();
+
+            setTimeout(function () {
+                assert.isFalse(xhr.triggerTimeout.called);
+                done();
+            }, 2);
+        });
+
+        it("does not get called after being aborted", function () {
+            this.sandbox.useFakeTimers();
+
+            this.xhr.open("GET", "/");
+            this.xhr.timeout = 1;
+            this.xhr.triggerTimeout = sinonSpy();
+            this.xhr.send();
+            this.xhr.abort();
+
+            this.sandbox.clock.tick(1);
+            assert.isFalse(this.xhr.triggerTimeout.called);
+        });
+
+        it("does not get called after erroring", function () {
+            this.sandbox.useFakeTimers();
+
+            this.xhr.open("GET", "/");
+            this.xhr.timeout = 1;
+            this.xhr.triggerTimeout = sinonSpy();
+            this.xhr.send();
+            this.xhr.error();
+
+            this.sandbox.clock.tick(1);
+            assert.isFalse(this.xhr.triggerTimeout.called);
+        });
+
+        it("only gets called with fake timers", function () {
```

Comment:
I don't really understand this restriction. Why would we want to restrict this functionality to fake timers - shouldn't this be up to the programmer if she want to control time?